base/src/main/java/io/spine/type/KnownTypes.java

```diff
@@ -269,6 +278,7 @@ public static void extendWith(TypeSet moreKnownTypes) {
             try {
                 TypeSet newKnownTypes = instance.typeSet.union(moreKnownTypes);
                 instance = new KnownTypes(newKnownTypes);
+                ValidationRules.updateFrom(instance.typeSet.messageTypes());
```

Comment:
I am not sure why we update the rules with all the message types every time. We could use just the message types from `moreKnownTypes` — and just add the newly discovered rules to the existing set.